kubeconfig/kubeconfig.go

```diff
@@ -0,0 +1,118 @@
+//
+// Copyright (c) 2019-2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package kubeconfig
+
+import (
+	"io/ioutil"
+	"net"
+	"os"
+
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+type KubeConfig struct {
+	APIVersion     string     `yaml:"apiVersion"`
+	Clusters       []Clusters `yaml:"clusters"`
+	Users          []Users    `yaml:"users"`
+	Contexts       []Contexts `yaml:"contexts"`
+	CurrentContext string     `yaml:"current-context"`
+	Kind           string     `yaml:"kind"`
+}
+
+type Clusters struct {
+	Cluster ClusterInfo `yaml:"cluster"`
+	Name    string      `yaml:"name"`
+}
+
+type ClusterInfo struct {
+	CertificateAuthority string `yaml:"certificate-authority"`
+	Server               string `yaml:"server"`
+}
+
+type Users struct {
+	Name string `yaml:"name"`
+	User User   `yaml:"user"`
+}
+
+type User struct {
+	Token string `yaml:"token"`
+}
+
+type Contexts struct {
+	Context Context `yaml:"context"`
+	Name    string  `yaml:"name"`
+}
+
+type Context struct {
+	Cluster   string `yaml:"cluster"`
+	Namespace string `yaml:"namespace"`
+	User      string `yaml:"user"`
+}
+
+func generateKubeConfig(token string) *KubeConfig {
+	host, port := os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT")
+	server := "https://" + net.JoinHostPort(host, port)
+	return &KubeConfig{
+		APIVersion: "v1",
+		Clusters: []Clusters{
+			Clusters{
+				Cluster: ClusterInfo{
+					CertificateAuthority: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+					Server:               server,
+				},
+				Name: server,
+			},
+		},
+		Users: []Users{
+			Users{
+				Name: "developer",
+				User: User{
+					Token: token,
+				},
+			},
+		},
+		Contexts: []Contexts{
+			Contexts{
+				Context: Context{
+					Cluster:   server,
+					Namespace: "che-workspace-controller",
```

Comment:
Do we really need hardcode namespace? We have https://github.com/eclipse/che-machine-exec/blob/master/exec/namespace_provider.go to get this information